extrollout/rollout_discovery.go

```diff
@@ -0,0 +1,216 @@
+	for _, rollout := range rollouts {
+		if client.IsExcludedFromDiscovery(metav1.ObjectMeta{
+			Name:        rollout.GetName(),
+			Namespace:   rollout.GetNamespace(),
+			Annotations: rollout.GetAnnotations(),
+			Labels:      rollout.GetLabels(),
+		}) {
+			continue
+		}
+		filteredRollouts = append(filteredRollouts, rollout)
+	}
+
+	targets := make([]discovery_kit_api.Target, len(filteredRollouts))
+	nodes := d.k8s.Nodes()
+
+	for i, rollout := range filteredRollouts {
+		targetName := fmt.Sprintf("%s/%s/%s", extconfig.Config.ClusterName, rollout.GetNamespace(), rollout.GetName())
+		attributes := map[string][]string{
+			"k8s.namespace":      {rollout.GetNamespace()},
+			"k8s.argo-rollout":   {rollout.GetName()},
+			"k8s.workload-type":  {"argo-rollout"},
+			"k8s.workload-owner": {rollout.GetName()},
+			"k8s.cluster-name":   {extconfig.Config.ClusterName},
+			"k8s.distribution":   {d.k8s.Distribution},
+		}
+
+		// Get replicas from spec
+		if replicas, found, err := unstructured.NestedInt64(rollout.Object, "spec", "replicas"); err == nil && found {
+			attributes["k8s.specification.replicas"] = []string{fmt.Sprintf("%d", replicas)}
+		}
+
+		// Get min-ready-seconds from spec
+		if minReadySeconds, found, err := unstructured.NestedInt64(rollout.Object, "spec", "minReadySeconds"); err == nil && found {
+			attributes["k8s.specification.min-ready-seconds"] = []string{fmt.Sprintf("%d", minReadySeconds)}
+		}
+
+		// Add labels
+		for key, value := range rollout.GetLabels() {
+			if !slices.Contains(extconfig.Config.LabelFilter, key) {
+				attributes[fmt.Sprintf("k8s.argo-rollout.label.%v", key)] = []string{value}
+				attributes[fmt.Sprintf("k8s.label.%v", key)] = []string{value}
+			}
+		}
+
+		extnamespace.AddNamespaceLabels(d.k8s, rollout.GetNamespace(), attributes)
+
+		// Get pod template labels
+		if podTemplate, found, err := unstructured.NestedMap(rollout.Object, "spec", "template", "metadata", "labels"); err == nil && found {
+			// Convert map[string]interface{} to map[string]string
+			podTemplateLabels := make(map[string]string)
+			for k, v := range podTemplate {
+				if str, ok := v.(string); ok {
+					podTemplateLabels[k] = str
+				}
+			}
+
+			// Get pods by label selector
+			pods := d.k8s.PodsByLabelSelector(&metav1.LabelSelector{
+				MatchLabels: podTemplateLabels,
+			}, rollout.GetNamespace())
+
+			// Add pod-based attributes
+			for key, value := range extcommon.GetPodBasedAttributes("argo-rollout", metav1.ObjectMeta{
+				Name:        rollout.GetName(),
+				Namespace:   rollout.GetNamespace(),
+				Annotations: rollout.GetAnnotations(),
+				Labels:      rollout.GetLabels(),
+			}, pods, nodes) {
+				attributes[key] = value
+			}
+
+			// Add service names
+			for key, value := range extcommon.GetServiceNames(d.k8s.ServicesMatchingToPodLabels(rollout.GetNamespace(), podTemplateLabels)) {
+				attributes[key] = value
+			}
+		}
+
+		// TODO: Get HPA if exists
```

Comment:
What's about these TODOs? Kube-Score will not show anything for Argo Rollouts. I guess we can remove these lines.